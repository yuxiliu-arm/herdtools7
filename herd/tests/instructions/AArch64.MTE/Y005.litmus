AArch64 Y005
(* IRG with all ones, forall *)
{
int x=100;
0:X0=x;
0:GCR_EL1=0xFFFFFFFFFFFF0000; (* RRND=1, Exclude=0 *)
}
 P0            ;
 MOV W2,#65535 ; (* 0xFFFF *)
 IRG X1,X0,X2  ;
forall 0:X1=x:t0

