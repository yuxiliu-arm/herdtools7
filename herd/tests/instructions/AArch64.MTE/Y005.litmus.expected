Test Y005 Required
States 1
0:X1=x:t0;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X1=x:t0)
Observation Y005 Always 1 0
Hash=6c900fc13a3b724a96e1ce198a96ba74

